### PR TITLE
Update UI to include Dashboard

### PR DESCRIPTION
**Pull request by amitha2210**
Updates the Ui to include a dashboard with summary of student numbers and tutorials offered

### CODE REVIEW

File: src/main/java/seedu/address/ui/Dashboard.java [hunk 1]
```diff
@@ -0,0 +1,146 @@
...
+    @FXML
+    private Label feesOverdue;
+    @FXML
+    private ScrollPane scrollPane;
+    @FXML
+    private HBox tutorials;
+
+
+    /**
+     * Creates a {@code Dashboard} with the given {@code Payment} t0 display.
```
Comment (gladwintan) [inline src/main/java/seedu/address/ui/Dashboard.java:44]: Typo in javadoc comment

File: src/main/java/seedu/address/ui/Dashboard.java [hunk 2]
```diff
@@ -0,0 +1,146 @@
...
+        addTutorialListListener();
+    }
+
+
+    private void initializeTutorialCards() {
+        for (Tutorial tutorial : tutorialList) {
+            List<Participation> initialParticipationList = getParticipationListForTutorial(tutorial);
+            TutorialCard card = new TutorialCard(tutorial.getSubject(), initialParticipationList);
+            tutorials.getChildren().add(card.getRoot());
+            tutorialCards.put(tutorial, card);
```
Comment (gladwintan) [inline src/main/java/seedu/address/ui/Dashboard.java:65]: I think we can use the createAndAddTutorial method here

File: src/main/java/seedu/address/ui/Dashboard.java [hunk 3]
```diff
@@ -0,0 +1,146 @@
...
+            }
+        }
+    }
+
+    private void updateStudentSummary() {
+        // Update total number of students
+        total.setText(String.valueOf(personList.size()));
+
+        // Calculate and update number of students with overdue fees
+        long studentsWithFeesOverdue = personList.stream()
```
Comment (gladwintan) [inline src/main/java/seedu/address/ui/Dashboard.java:139]: Would int be better here?

File: src/main/java/seedu/address/ui/TutorialCard.java [hunk 4]
```diff
@@ -0,0 +1,54 @@
...
+ */
+public class TutorialCard extends UiPart<Region> {
+
+    private static final String FXML = "TutorialCard.fxml";
+
+    @FXML
+    private Label tutorial;
+    @FXML
+    private Label studentsEnrolled;
+    private List<Participation> participationList;
```
Comment (gladwintan) [inline src/main/java/seedu/address/ui/TutorialCard.java:24]: Should this be reordered for readability?

File: src/main/java/seedu/address/ui/TutorialCard.java [hunk 5]
```diff
@@ -0,0 +1,54 @@
...
+     */
+    public void updateParticipationList(List<Participation> newParticipationList) {
+        this.participationList = newParticipationList;
+        updateStudentCount();
+    }
+    private void updateStudentCount() {
+        this.studentsEnrolled.setText(String.valueOf(participationList.size()));
+    }
+
+    public VBox getRoot() {
```
Comment (gladwintan) [inline src/main/java/seedu/address/ui/TutorialCard.java:50]: Is this required?

**Review by gladwintan — COMMENTED** [review]
nit

**Review by gladwintan — APPROVED** [review]
LGTM

File: src/main/java/seedu/address/model/Model.java [hunk 6]
```diff
@@ -86,6 +86,9 @@ public interface Model {
     /** Returns an unmodifiable view of the filtered person list */
     ObservableList<Person> getFilteredPersonList();
 
+    /** Returns an unmodifiable view of the entire person list */
```
Comment (Slipperybara) [inline src/main/java/seedu/address/model/Model.java:89]: Nice implementation

**Review by Slipperybara — APPROVED** [review]
LGTM

**Pull request merged**